plugins/BEdita/Core/src/Model/Action/AddAssociatedAction.php

```diff
@@ -44,7 +44,20 @@ protected function diff(EntityInterface $entity, array $relatedEntities)
         foreach ($relatedEntities as $relatedEntity) {
             $primaryKey = $relatedEntity->extract($bindingKey);
             if (in_array($primaryKey, $existing)) {
-                continue;
+                if (!($this->Association instanceof BelongsToMany) || !$relatedEntity->has('_joinData')) {
+                    continue;
+                }
+
+                $relatedEntity->_joinData = $this->Association->junction()->patchEntity(
+                    $this->Association->junction()->find()
+                        ->where([
+                            $this->Association->getForeignKey() => $entity->get($this->Association->getSource()->getPrimaryKey()),
```

Comment:
again...

Reply:
No, sorry. My previous comment is wrong, and you were right.
It would be better to handle the case where there are composite primary/foreign keys. 😱

Reply:
God forgives, Scrutinizer doesn't